### DISCUSSION TITLE
Why my project is slow during development? 😞

### DISCUSSION THREAD

**Discussion by abhagsain** [vercel/next.js#12711] [Help]
I only have two pages in my project but the hot reloads takes forever to load.
This development experience is taking me back to my Android Development days 

Also I'm using Redux. Is that causing the app slow build? 
Thank you for your help.

This is my package.json

```
{
  "name": "jsmtc",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "dev": "next",
    "build": "next build",
    "start": "next start",
    "clean": "next clean",
    "plop": "plop",
    "storybook": "start-storybook -p 6006",
    "build-storybook": "build-storybook"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@tailwindcss/ui": "^0.2.2",
    "@zeit/next-css": "^1.0.1",
    "@zeit/next-sass": "^1.0.1",
    "autoprefixer": "^9.7.4",
    "axios": "^0.19.2",
    "immer": "^6.0.5",
    "next": "^9.3.0",
    "prop-types": "^15.7.2",
    "react": "^16.13.0",
    "react-dom": "^16.13.0",
    "react-modal": "^3.11.2",
    "react-redux": "^7.2.0",
    "react-syntax-highlighter": "^12.2.1",
    "redux": "^4.0.5",
    "redux-devtools-extension": "^2.13.8",
    "redux-thunk": "^2.3.0",
    "tailwindcss": "^1.2.0"
  },
  "devDependencies": {
    "@babel/core": "^7.9.6",
    "@storybook/addon-actions": "^5.3.18",
    "@storybook/addon-links": "^5.3.18",
    "@storybook/addons": "^5.3.18",
    "@storybook/react": "^5.3.18",
    "babel-loader": "^8.1.0",
    "eslint": "^6.8.0",
    "eslint-plugin-react": "^7.19.0",
    "plop": "^2.6.0"
  }
}

```

And this is my next.config.js

```
const withCSS = require("@zeit/next-css");
module.exports = withCSS({
  cssLoaderOptions: {
    url: false,
  }
});
```

**Comment by marcusmota** [comment]
We have the same issue here.

even when importing only one function (can be actions, reducers, selectors, requests) the page development build have the whole imported file on its scope (recursively 

e.g if we have another function inside of fetchUser, it imports its whole children file as well) 

```
import { fetchUser } from './src/requests/user';

...
somewhere inside the component

fetchUser();  //this will import the whole file to the build recursively (if I import another function from another file in fetchUser() the same will happen and so on)

...

```

Not sure If I'm doing something wrong, but this increase the page build time in development a lot and also its bundle size.

Thanks

**Reply by abhagsain** [reply]
I'm using Storybook along side next.js. Is that increasing the load time? 
This is really really frustrating 😪

![tree](https://user-images.githubusercontent.com/36589645/81560429-045d3a80-93af-11ea-959e-4b12d6d2c37c.png)

**Reply by Akumzy** [reply]
@abhagsain Off topic but please what tool did you use to list the folder structure like that?

**Reply by abhagsain** [reply]
This CLI [tree-node-cli](https://github.com/yangshun/tree-node-cli)
However for some reason it's not working on Windows command line so you need to run it on Bash cmd

```npm install tree-node-cli```

**Reply by Akumzy** [reply]
Okay thanks, though I am not using Windows

**Reply by jibin2706** [reply]
> We have the same issue here.
> 
> even when importing only one function (can be actions, reducers, selectors, requests) the page development build have the whole imported file on its scope (recursively
> 
> e.g if we have another function inside of fetchUser, it imports its whole children file as well)
> 
> ```
> import { fetchUser } from './src/requests/user';
> 
> ...
> somewhere inside the component
> 
> fetchUser();  //this will import the whole file to the build recursively (if I import another function from another file in fetchUser() the same will happen and so on)
> 
> ...
> ```
> 
> Not sure If I'm doing something wrong, but this increase the page build time in development a lot and also its bundle size.
> 
> Thanks

Even I'm facing the same issues

**Comment by Timer** [comment]
Without a full reproducible demo, there's no way for us to know what is wrong with your project.

**Reply by abhagsain** [reply]
Hi @Timer **I'm using Storybook with Next.js and it's in the same folder with my other .js files.** 

Do you think that's causing it to load slow? I don't have any custom webpack configuration. 

![jsmtc](https://user-images.githubusercontent.com/36589645/81590889-2966a300-93d9-11ea-8444-bd819b69eb27.png)

**Reply by Timer** [reply]
This would only be slow if they were in the `pages/` directory.

**Reply by abhagsain** [reply]
> This would only be slow if they were in the `pages/` directory.

@Timer If I share my project can you please figure out why it's taking longer to build pages during development. It takes ~10 sec even if I change one line. It's slowing down the development process 😥

**Reply by Timer** [reply]
Yes, sharing your project would really help.

**Comment by threehams** [comment]
Tailwind can be very slow in dev mode, especially with many combinations enabled, unless you're using the JIT compiler.

https://tailwindcss.com/docs/just-in-time-mode

edit: _wow_ I didn't read the year on this issue.
